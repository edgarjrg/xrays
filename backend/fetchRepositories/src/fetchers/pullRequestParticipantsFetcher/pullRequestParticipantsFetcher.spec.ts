import { findAuthor, findApprovers } from './pullRequestParticipantsFetcher';
import { ap } from 'ramda';
// import * as Bluebird from 'bluebird';

const pullRequestLink = {
  "projectKey": "atlassian-oauth",
  "pullRequestURL": "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/24",
  "repoSlug": "atlassian-oauth",
  "userName": "atlassian",
}

describe('finders', () => {
  const firstPage = require('./testData/firstPage.json')
  const secondPage = require('./testData/secondPage.json')

  test('findAuthor should get author', async () => {
    expect(ap([findAuthor], [firstPage, secondPage])).toMatchSnapshot()
  });

  test('findApprovers should get approvers', async () => {
    expect(ap([findApprovers], [firstPage, secondPage])).toMatchSnapshot()
  });

  test('', () => {
    expect(ap([findAuthor, findApprovers], [firstPage, secondPage])).toMatchSnapshot()
  })
});
