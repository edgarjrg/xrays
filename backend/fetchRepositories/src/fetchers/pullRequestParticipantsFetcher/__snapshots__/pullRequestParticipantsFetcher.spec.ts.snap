// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finders  1`] = `
Array [
  undefined,
  Object {
    "author": "https://api.bitbucket.org/2.0/users/ppetrowski",
    "date": "2017-02-27T15:31:22.497552+00:00",
    "pullRequestURL": "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/34",
    "type": "author",
  },
  Array [
    Object {
      "author": "https://api.bitbucket.org/2.0/users/lpater",
      "date": "2017-03-07T15:11:57.673660+00:00",
      "pullRequestURL": "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/34",
      "type": "participant",
    },
  ],
  Array [
    Object {
      "author": "https://api.bitbucket.org/2.0/users/mmakowskii",
      "date": "2017-02-28T07:14:07.634137+00:00",
      "pullRequestURL": "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/34",
      "type": "participant",
    },
  ],
]
`;

exports[`finders findApprovers should get approvers 1`] = `
Array [
  Array [
    Object {
      "author": "https://api.bitbucket.org/2.0/users/lpater",
      "date": "2017-03-07T15:11:57.673660+00:00",
      "pullRequestURL": "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/34",
      "type": "participant",
    },
  ],
  Array [
    Object {
      "author": "https://api.bitbucket.org/2.0/users/mmakowskii",
      "date": "2017-02-28T07:14:07.634137+00:00",
      "pullRequestURL": "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/34",
      "type": "participant",
    },
  ],
]
`;

exports[`finders findAuthor should get author 1`] = `
Array [
  undefined,
  Object {
    "author": "https://api.bitbucket.org/2.0/users/ppetrowski",
    "date": "2017-02-27T15:31:22.497552+00:00",
    "pullRequestURL": "https://api.bitbucket.org/2.0/repositories/atlassian/atlassian-oauth/pullrequests/34",
    "type": "author",
  },
]
`;
